spk/salt-minion/src/service-setup.sh

```diff
@@ -1,33 +1,37 @@
 
 service_postinst ()
 {
     # Create a Python virtualenv
     ${VIRTUALENV} --system-site-packages ${SYNOPKG_PKGDEST}/env
 
-    # Install wheels
+    # Install the wheels
     wheelhouse=${SYNOPKG_PKGDEST}/share/wheelhouse
-    ${SYNOPKG_PKGDEST}/env/bin/pip install --no-deps --force-reinstall --no-index --find-links ${wheelhouse} ${wheelhouse}/*.whl
+    ${SYNOPKG_PKGDEST}/env/bin/pip install --no-deps --no-index --upgrade --force-reinstall -f ${wheelhouse} ${wheelhouse}/*.whl
 
-    # Patch rsax931.py file to find libcrypto lib
-    # (Rely on patch util bundled with python3's busybox)
-    ${PYTHON_DIR}/bin/patch ${SYNOPKG_PKGDEST}/env/lib/python3.7/site-packages/salt/utils/rsax931.py < ${SYNOPKG_PKGDEST}/share/rsax931.py.patch
+    # Prepare salt-minion config in /var/packages/salt-minion/
+    test -L ${SYNOPKG_PKGDEST}/env/etc || ln -s /var/packages/${SYNOPKG_PKGNAME}/etc ${SYNOPKG_PKGDEST}/env/etc
+    test -d ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d || install -m 755 -o sc-${SYNOPKG_PKGNAME} -g ${SYNOPKG_PKGNAME} -d ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion || install -m 644 -o sc-${SYNOPKG_PKGNAME} -g ${SYNOPKG_PKGNAME} ${SYNOPKG_PKGDEST}/share/minion ${SYNOPKG_PKGDEST}/env/etc/salt/minion
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/proxy || install -m 644 -o sc-${SYNOPKG_PKGNAME} -g ${SYNOPKG_PKGNAME} ${SYNOPKG_PKGDEST}/share/proxy ${SYNOPKG_PKGDEST}/env/etc/salt/proxy
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/02_pidfile.conf || echo "pidfile: ${PID_FILE}" > ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/02_pidfile.conf
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/01_rootdir.conf || echo "root_dir: ${SYNOPKG_PKGDEST}/env" > ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/01_rootdir.conf
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/03_logging.conf || echo "log_file: udp://localhost:10514" > ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/03_logging.conf
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/03_logging.conf || echo "log_level_logfile: info" >> ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/03_logging.conf
 
-    # Prepare salt-minion config in /var/salt
-    install -m 755 -d ${SYNOPKG_PKGDEST}/var
-    install -m 755 -d ${SYNOPKG_PKGDEST}/var/minion.d
-    install -m 644 ${SYNOPKG_PKGDEST}/share/minion.conf ${SYNOPKG_PKGDEST}/var
-    echo "pidfile: ${PID_FILE}" > ${SYNOPKG_PKGDEST}/var/minion.d/02_pidfile.conf
     # Populate salt master address and minion_id only if file don't already exist
-    test -f ${SYNOPKG_PKGDEST}/var/minion.d/99-master-address.conf || echo "master: salt" > ${SYNOPKG_PKGDEST}/var/minion.d/99-master-address.conf
-    test -f ${SYNOPKG_PKGDEST}/var/minion.d/98-minion-id.conf || echo "id: myname" > ${SYNOPKG_PKGDEST}/var/minion.d/98-minion-id.conf
-}
+    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/99-master-address.conf || echo "master: salt" > ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/99-master-address.conf
```

Comment:
Should master really be salt? Not 127.0.0.1 or localhost?

Reply:
Is the file needed? Isn't the default salt anyway? Sorry for the noob questions.

Reply:
Thanks, then I suggest removing or changing it to localhost since that IMHO is a more reasonable default. People can/will change it anyway.

```suggestion
    test -f ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/99-master-address.conf || echo "master: localhost" > ${SYNOPKG_PKGDEST}/env/etc/salt/minion.d/99-master-address.conf
```